src/test/spec/unified_runner/observer.rs

```diff
@@ -0,0 +1,103 @@
+use tokio::sync::{
+    broadcast::{
+        self,
+        error::{RecvError, TryRecvError},
+    },
+    RwLock,
+};
+
+use std::{sync::Arc, time::Duration};
+
+use crate::{
+    error::{Error, Result},
+    runtime,
+    test::Event,
+};
+
+use super::{events_match, EntityMap, ExpectedEvent};
+
+/// Observer used to cache all the seen events for a given client in a unified test.
+/// Used to implement assertEventCount and waitForEvent operations.
+#[derive(Debug)]
+pub(crate) struct EventObserver {
```

Comment:
The existing `EventHandler` "consumed" events as they were observed, which doesn't work for the functionality we need in the unified runner. Down the road, we may want to convert all the usages of `EventHandler` to use `EventObserver` so we can ditch the other type, but I figured that was out of scope for this work.

Reply:
I'm having trouble following the distinction between this and `EventHandler` - can you explain more about what was needed that couldn't be done using that?

Reply:
Got it, thank you!  Can you leave a TODO in the code pointing to a new ticket so we don't lose track of this?

Reply:
Filed RUST-1425, added comment.